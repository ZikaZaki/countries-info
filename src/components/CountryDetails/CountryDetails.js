import React from 'react';
import { useSelector } from 'react-redux';
import { HiOutlineArrowCircleRight } from 'react-icons/hi';

// Import Styling
import styles from './CountryDetails.module.css';

const CountryDetails = () => {
  const countryDetails = useSelector((state) => state.details);

  return (
    <div className={styles['details-container']} data-testid="details-container">
      <div className={styles['country-map']}>
        <img src={countryDetails.flag} alt="Country Map" />
      </div>
      <div className={styles.headline}>
        <h2>
          {countryDetails.continent}
          /
          {countryDetails.commonName}
          {' '}
          Details
        </h2>
      </div>
      <div className={styles['details-list']}>
        <div className={styles['list-item']}>
          <h2>
            Common Name:
            {countryDetails.commonName}
          </h2>
          <HiOutlineArrowCircleRight />
        </div>
        <div className={styles['list-item']}>
          <h2>
            Official Name:
            {countryDetails.officialName}
          </h2>
          <HiOutlineArrowCircleRight />
        </div>
        <div className={styles['list-item']}>
          <h2>
            Capital:
            {countryDetails.capital}
          </h2>
          <HiOutlineArrowCircleRight />
        </div>
        <div className={styles['list-item']}>
          <h2>
            Population:
            {countryDetails.population}
          </h2>
          <HiOutlineArrowCircleRight />
        </div>
        <div className={styles['list-item']}>
          <h2>
            Area:
            {countryDetails.area}
          </h2>
          <HiOutlineArrowCircleRight />
        </div>
        <div className={styles['list-item']}>
          <h2>
            Timezone:
            {countryDetails.timezone}
          </h2>
          <HiOutlineArrowCircleRight />
        </div>
      </div>
    </div>
  );
};

export default CountryDetails;
