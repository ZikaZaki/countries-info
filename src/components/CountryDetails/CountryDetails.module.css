.details-container {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  font-family: var(--lato);
  min-height: 100vh;
  width: 100%;

  /* background-color: var(--primary); */
  background-color: #ff5294;
  color: var(--neutral);
}

.country-map {
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 40vh;
  width: 100%;
  padding: 1rem;
}

.country-map img {
  display: flex;
  width: 100%;
  height: 100%;
  object-fit: contain;
  padding: 0;
}

.headline {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  vertical-align: middle;
  gap: 0.5rem;
  height: 5vh;
  width: 100%;
  padding: 0.2rem 1rem;
  background-color: var(--primary);
}

.headline h2 {
  font-size: calc(8px + 1vmin);
  font-weight: bold;
  text-align: center;
  padding: 0;
  margin: 0;
}

.details-list {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  padding: 0;
}

.list-item {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 5vh;
  width: 100%;
  padding: 2rem;
}

.list-item h2 {
  display: flex;
  font-size: calc(10px + 2vmin);
  font-weight: bold;
  text-align: center;
  padding: 0;
  margin: 0;
}

.list-item:nth-child(even) {
  background-color: var(--primary);
}

.list-item svg {
  display: flex;
  font-size: calc(20px + 2vmin);
  font-weight: bold;
  text-align: center;
  justify-self: flex-end;
  align-self: center;
  padding: 0;
  margin: 0;
}
