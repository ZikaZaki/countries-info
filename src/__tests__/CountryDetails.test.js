import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import CountryDetails from '../components/CountryDetails/CountryDetails';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', async () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <CountryDetails />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the component container is there', async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <CountryDetails />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('details-container');
  expect(container).toBeInTheDocument();
});
