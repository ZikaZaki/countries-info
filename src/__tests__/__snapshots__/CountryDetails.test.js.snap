// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
<div
  className="details-container"
  data-testid="details-container"
>
  <div
    className="country-map"
  >
    <img
      alt="Country Map"
    />
  </div>
  <div
    className="headline"
  >
    <h2>
      /
       
      Details
    </h2>
  </div>
  <div
    className="details-list"
  >
    <div
      className="list-item"
    >
      <h2>
        Common Name:
      </h2>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="list-item"
    >
      <h2>
        Official Name:
      </h2>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="list-item"
    >
      <h2>
        Capital:
      </h2>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="list-item"
    >
      <h2>
        Population:
      </h2>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="list-item"
    >
      <h2>
        Area:
      </h2>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="list-item"
    >
      <h2>
        Timezone:
      </h2>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
  </div>
</div>
`;
